src/main.py

```diff
@@ -1,125 +1,250 @@
+from importlib import reload
 from typing import Optional
+from utils import check_roles, format_roles_error
 
-class Leah:
+
+# Bot definition
+class Leah(commands.Bot):
     def __init__(self):
-        self.posted = set()
+        super().__init__(
+            command_prefix=COMMAND_PREFIX,
+            intents=DISCORD_INTENTS,
+            description=strings.get("client_description"),
+            allowed_mentions=discord.AllowedMentions.none())
+
+    async def sync_guild(self, guild: discord.Guild):
+        self.tree.copy_global_to(guild=guild)
+        await self.tree.sync(guild=guild)
+
+    async def setup_hook(self) -> None:
+        # Load all extensions on setup
+        for ext in EXTENSIONS:
+            await self.load_extension(name=ext)
+
+    async def on_ready(self):
+        print(strings.get("client_login").format(bot.user.name, bot.user.discriminator, bot.user.id))
 
-leah = Leah()
+    def reload_strings(self):
+        reload(strings)
 
-@client.event
-async def on_ready():
-    print("Logged in as:")
-    print(client.user.name)
-    print(client.user.id)
 
-@client.event
-async def on_reaction_add(reaction: discord.Reaction, user: discord.User):
-    # Ignore messages that have been handled previously
-    if reaction.message.id in leah.posted:
+bot = Leah()
+
+
+@bot.listen(name="on_message")
+async def on_message(message: discord.Message) -> None:
+    """
+    For messages in listening channels, reposts a preview to the verification channel.
+    """
+    # Ignore all bots
+    if message.author.bot:
         return
 
-    # Staff reactions to verification channel posts by this client
-    if reaction.message.channel.id == VERIFY_CHAN:
-        # Ignore reactions to posts not by this client in verification channel
-        if client.user != reaction.message.author:
-            return
+    # Send posts from listening channels to verification channel
+    if message.channel.id in ART_CHANS:
+        await verify_art(message=message)
+
+@bot.tree.context_menu(name=strings.get("app_name_publish"), guilds=GUILDS)
+async def command_publish(interaction: discord.Interaction, message: discord.Message):
+    """
+    Reposts a message from a self-curated or verification channel as a formatted embed to a showcase or gallery channel,
+    respectively, depending on the channel the original message was posted in.
+    Posts in showcase channels may also be removed with this command.
+    """
+    reply = None
+    success = False      # Reactions are added to messages based on success or failure
+    fail_quietly = True  # Failure reactions are hidden by default
+    reaction_users = []
+    for reaction in message.reactions:
+        reaction_users.append([user async for user in reaction.users()])
+    posted = bot.user in sum(reaction_users, [])  # We avoid duplicate publish actions by checking for reactions
+
+    # Interactions on verification channel posts by this bot
+    if message.channel.id == VERIFY_CHAN:
```

Comment:
It's definitely quite a long method with a lot of branches, but it might be annoying to return `(success, reply, fail_quietly)` for every method unless you're fine with that

Reply:
That's a fair point... maybe some day it would be nice to refactor, but we can leave it as is for now.